platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/TipsetEventCreator.java

```diff
@@ -0,0 +1,371 @@
+        tipsetMetrics.getBullyScoreMetric().update(bullyScore);
+
+        // Never bother with anti-bullying techniques if we have a bully score of 1. We are pretty much guaranteed
+        // to bully ~1/3 of other nodes by a score of 1.
+        final double beNiceToNerdChance = (bullyScore - 1) / antiBullyingFactor;
+
+        if (beNiceToNerdChance > 0 && random.nextDouble() < beNiceToNerdChance) {
+            return createEventToReduceBullyScore();
+        } else {
+            return createEventByOptimizingTipsetScore();
+        }
+    }
+
+    /**
+     * Create an event using the other parent with the best tipset score.
+     *
+     * @return the new event, or null if it is not legal to create a new event
+     */
+    @Nullable
+    private GossipEvent createEventByOptimizingTipsetScore() {
+        final List<EventDescriptor> possibleOtherParents = childlessOtherEventTracker.getChildlessEvents();
+        Collections.shuffle(possibleOtherParents, random);
+
+        EventDescriptor bestOtherParent = null;
+        long bestScore = 0;
+        for (final EventDescriptor otherParent : possibleOtherParents) {
+            final long parentScore = tipsetScoreCalculator.getTheoreticalAdvancementScore(List.of(otherParent));
+            if (parentScore > bestScore) {
+                bestOtherParent = otherParent;
+                bestScore = parentScore;
+            }
+        }
+
+        if (lastSelfEvent != null && bestOtherParent == null) {
+            // There exist no parents that can advance consensus, and this is not our first event.
+            return null;
+        }
+
+        return buildEvent(bestOtherParent);
+    }
+
+    /**
+     * Create an event that reduces the bully score.
+     *
+     * @return the new event, or null if it is not legal to create a new event
+     */
+    @Nullable
+    private GossipEvent createEventToReduceBullyScore() {
+        final List<EventDescriptor> possibleOtherParents = childlessOtherEventTracker.getChildlessEvents();
+        final List<EventDescriptor> nerds = new ArrayList<>(possibleOtherParents.size());
+
+        // Choose a random nerd, weighted by how much it is currently being bullied.
+
+        // First, figure out who is a nerd and sum up all bully scores.
+        int bullyScoreSum = 0;
+        final List<Integer> bullyScores = new ArrayList<>(possibleOtherParents.size());
+        for (final EventDescriptor nerd : possibleOtherParents) {
+            final int nodeIndex = addressBook.getIndexOfNodeId(nerd.getCreator());
+            final int bullyScore = tipsetScoreCalculator.getBullyScoreForNodeIndex(nodeIndex);
+
+            final long tipsetScore = tipsetScoreCalculator.getTheoreticalAdvancementScore(List.of(nerd));
+
+            if (bullyScore > 1 && tipsetScore > 0) {
+                nerds.add(nerd);
+                bullyScores.add(bullyScore);
+                bullyScoreSum += bullyScore;
+            }
+        }
+
+        if (nerds.isEmpty()) {
```

Comment:
Added comment. Also, I have this block return `null`, since it makes it simpler when tracing code pathways not to have to consider multiple pathways for `createEventByOptimizingTipsetScore()`.

```
        if (nerds.isEmpty()) {
            // Note: this should be impossible, since we will not enter this method in the first
            // place if there are no nerds. But better to be safe than sorry, and returning null
            // is an acceptable way of saying "I can't create an event right now".
            return null;
        }
```